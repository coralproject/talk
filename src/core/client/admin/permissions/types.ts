import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
