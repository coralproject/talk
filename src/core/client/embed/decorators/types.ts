import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
