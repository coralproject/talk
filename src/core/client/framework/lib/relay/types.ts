import { _RefType } from "react-relay";

export type FragmentKeys<T> = {
  [P in keyof T]: T[P] extends _RefType<any> | null ? P : never
}[keyof T];

export type FragmentKeysNoLocal<T> = Exclude<FragmentKeys<T>, "local">;
