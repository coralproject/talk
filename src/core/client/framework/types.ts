// TODO: (@cvle) Extract useful common types into its own package.
export { Diff, Omit, Overwrite, PropTypesOf } from "talk-ui/types";
