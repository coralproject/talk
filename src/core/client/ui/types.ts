import React from "react";

// TODO: Extract useful common types into its own package.

/**
 * Overwrite properties of `T`.
 *
 * E.g. Omit<{a: boolean, b: boolean}, "b"> = {a: boolean}
 */
export type Omit<U, K extends keyof U> = Pick<U, Exclude<keyof U, K>>;

/**
 * Overwrite properties of `T`.
 *
 * E.g. Overwrite<{a: boolean}, {a: string}> = {a: string}
 */
export type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;

/**
 * Returns the PropTypes from a React Component.
 *
 * E.g. type ButtonProps = ReturnPropTypes<Button>;
 */
export type PropTypesOf<T> = T extends React.ComponentType<infer R>
  ? R
  : T extends React.Component<infer S> ? S : {};
