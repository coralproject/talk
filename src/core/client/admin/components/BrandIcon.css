.base {
}

.md {
  height: 35px;
  width: 36px;
}

.lg {
  height: 58px;
  width: 59px;
}
