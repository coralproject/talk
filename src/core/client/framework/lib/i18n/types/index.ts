export { default as FluentShortNumber } from "./FluentShortNumber";
