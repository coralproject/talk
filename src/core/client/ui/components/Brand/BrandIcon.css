.base {
}

.md {
  height: 25px;
  width: 25px;
}

.lg {
  height: 58px;
  width: 59px;
}
