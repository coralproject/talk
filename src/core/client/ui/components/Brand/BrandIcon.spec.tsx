import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import BrandIcon from "./BrandIcon";

it("renders correctly", () => {
  const props: PropTypesOf<typeof BrandIcon> = {
    className: "custom",
    size: "lg",
  };
  const wrapper = shallow(<BrandIcon {...props} />);
  expect(wrapper).toMatchSnapshot();
});
