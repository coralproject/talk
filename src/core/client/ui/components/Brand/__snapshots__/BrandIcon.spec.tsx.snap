// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrandIcon
  className="custom"
  classes={
    Object {
      "base": "BrandIcon-base",
      "lg": "BrandIcon-lg",
      "md": "BrandIcon-md",
    }
  }
  size="lg"
/>
`;
